src/lauxlib.js

```diff
@@ -644,19 +644,72 @@ const luaL_unref = function(L, t, ref) {
     }
 };
 
-// Only with Node
-if (!WEB) {
-    const fs = require('fs');
 
-    class LoadF {
-        constructor() {
-            this.n = NaN;  /* number of pre-read characters */
-            this.f = null;  /* file being read */
-            this.buff = new Buffer(1024);  /* area for reading file */
-            this.pos = 0;  /* current position in file */
-            this.err = void 0;
-        }
+const errfile = function(L, what, fnameindex, error) {
+    let serr = error.message;
+    let filename = lua.lua_tostring(L, fnameindex).slice(1);
+    lua.lua_pushstring(L, lua.to_luastring(`cannot ${what} ${lua.to_jsstring(filename)}: ${serr}`));
+    lua.lua_remove(L, fnameindex);
+    return lua.LUA_ERRFILE;
+};
+
+let getc = function() {
+    return null;
+};
+
+const utf8_bom = [0XEF, 0XBB, 0XBF];  /* UTF-8 BOM mark */
+const skipBOM = function(lf) {
+    lf.n = 0;
+    let c;
+    let p = 0;
+    do {
+        c = getc(lf);
+        if (c === null || c !== utf8_bom[p]) return c;
+        p++;
+        lf.buff[lf.n++] = c;  /* to be read by the parser */
+    } while (p < utf8_bom.length);
+    lf.n = 0;  /* prefix matched; discard it */
+    return getc(lf);  /* return next character */
+};
+
+/*
+** reads the first character of file 'f' and skips an optional BOM mark
+** in its beginning plus its first line if it starts with '#'. Returns
+** true if it skipped the first line.  In any case, '*cp' has the
+** first "valid" character of the file (after the optional BOM and
+** a first-line comment).
+*/
+const skipcomment = function(lf) {
+    let c = skipBOM(lf);
+    if (c === '#'.charCodeAt(0)) {  /* first line is a comment (Unix exec. file)? */
+        do {  /* skip first line */
+            c = getc(lf);
+        } while (c && c !== '\n'.charCodeAt(0));
+
+        return {
+            skipped: true,
+            c: getc(lf)  /* skip end-of-line, if present */
+        };
+    } else {
+        return {
+            skipped: false,
+            c: c
+        };
+    }
+};
+
+class LoadF {
+    constructor() {
+        this.n = NaN;  /* number of pre-read characters */
+        this.f = null;  /* file being read */
+        this.buff = new Buffer(1024);  /* area for reading file */
```

Comment:
We should only initialise this in the `!WEB` case: otherwise this will pull in a browser shim for node's Buffer type.